changelog.md

```diff
@@ -93,6 +93,7 @@
 
 - Renamed `-d:nimCompilerStackraceHints` to `-d:nimCompilerStacktraceHints`.
 
+- Now parsexxxInt, parsexxx and fromxxx(xxx stands for Hex, Bin, Oct) check overflow. Use `-d:nimLegacyParseInts` to switch to old behaviour.
```

Comment:
spell them all out so that a user grepping for parseHexInt (etc) will show this

Reply:
Make sense

Reply:
Since policy changed, what should we do now?
https://github.com/nim-lang/Nim/pull/18468

Reply:
> Since policy changed, what should we do now? Add it to strbasics instead?

I propose to add it to a separate Nimble package under nim-lang/, we probably need to review how to easily setup CI for a new nim-lang package first.

Reply:
Now I use `nimPreviewParseInt` to enable this feature.

Reply:
separate package doesn't cut it, we should either fix the bugs as done in this PR via nimPreviewParseInt, or add `parseNum` https://github.com/nim-lang/Nim/pull/18386#issuecomment-871230257

note that  `parseNum` should be added regardless, eventually; having a single generic API (with optional params) is a better design than the current situation with 9 (!) APIs:
parseHexInt/parseOctInt/parseBinInt, parseHex/parseHex/parseBin and fromHex/fromOct/fromBin

Reply:
> note that parseNum should be added regardless, eventually; having a single generic API (with optional params) is a better design than the current situation with 9 (!) APIs:
parseHexInt/parseOctInt/parseBinInt, parseHex/parseHex/parseBin and fromHex/fromOct/fromBin

Yes, agreed. And then we shouldn't fix the older procs.

Reply:
Ok